From 0ccdf623928dfe1ce04aa7a8995a5403feaa8a5e Mon Sep 17 00:00:00 2001
From: Mark Abraham <mark.j.abraham@gmail.com>
Date: Thu, 13 Oct 2022 18:39:00 +0000
Subject: [PATCH] Relax arbitrary test tolerances.

This is now in line with the fix also made for #3836

Fixes #4290
---
 src/programs/mdrun/tests/energycomparison.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/programs/mdrun/tests/energycomparison.cpp b/src/programs/mdrun/tests/energycomparison.cpp
index 940f2f9d69..2d25b2fe02 100644
--- a/src/programs/mdrun/tests/energycomparison.cpp
+++ b/src/programs/mdrun/tests/energycomparison.cpp
@@ -66,8 +66,8 @@ namespace test
 EnergyTermsToCompare EnergyComparison::defaultEnergyTermsToCompare()
 {
     return {
-        { interaction_function[F_EPOT].longname, relativeToleranceAsUlp(10.0, 50) },
-        { interaction_function[F_EKIN].longname, relativeToleranceAsUlp(10.0, 50) },
+        { interaction_function[F_EPOT].longname, relativeToleranceAsPrecisionDependentUlp(60.0, 200, 160) },
+        { interaction_function[F_EKIN].longname, relativeToleranceAsPrecisionDependentUlp(60.0, 200, 160) },
         // The pressure is very strongly affected by summation errors,
         // so we need a large tolerance.
         // The value of 17000 is calibrated for running a small water box for 16 steps.
-- 
GitLab

